inotify.go

```diff
@@ -36,7 +37,7 @@ type Watcher struct {
 // NewWatcher establishes a new watcher with the underlying OS and begins waiting for events.
 func NewWatcher() (*Watcher, error) {
 	// Create inotify fd
-	fd, errno := unix.InotifyInit()
+	fd, errno := unix.InotifyInit1(syscall.IN_CLOEXEC)
```

Comment:
We need to be careful with this change, possibly testing it manually as Travis CI doesn't give us coverage of `arm` and `ppc64` architectures that fsnotify currently supports.